### DISCUSSION TITLE
Disallow passing keyword arguments not in class?

### DISCUSSION THREAD

**Discussion by MartinBernstorff** [pydantic/pydantic#4660] [Question]
Hi guys!

A minimal example. We have a specification with:

```python
class Spec(BaseModel):
    kwargs: Optional[dict] = None
```

I had a type, where I tried initialising it like:

```python
Spec(wrong_kwargs_attribute={})
```

This didn't raise an error, but allowed the script to continue, ignoring the `wrong_kwargs_attribute.` Is there any way to have this raise an error? :-)

**Comment by willyd** [comment]
@MartinBernstorff 

You can do:
```python
from typing import Optional
from pydantic import BaseModel, Extra

class Spec(BaseModel, extra=Extra.forbid):
    kwargs: Optional[dict] = None
```
See the relevant docs: https://pydantic-docs.helpmanual.io/usage/model_config/

**Reply by MartinBernstorff** [reply]
Thanks a ton, Willy! I'd petition to make this default, but realise this might cause a bunch of back-compatability issues.

**Reply by samuelcolvin** [reply]
I understand it might seem odd, but in the majority of situations, ignoring extra values is the best default.
